xmtp_mls/src/groups/mod.rs

```diff
@@ -326,8 +331,12 @@ impl MlsGroup {
         let welcome = deserialize_welcome(&welcome_bytes)?;
 
         let join_config = build_group_join_config();
-        let staged_welcome =
-            StagedWelcome::new_from_welcome(provider, &join_config, welcome.clone(), None)?;
+        let processed_welcome =
+            ProcessedWelcome::new_from_welcome(provider, &join_config, welcome.clone())?;
+        if processed_welcome.psks().is_empty() {
+            return Err(GroupError::NoPSKSupport);
+        }
```

Comment:
We may eventually want to allow external psks depending on how we want to do conversation history. But in the meantime I'm okay with blocking both resumption and external psks.